subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyHandler.java

```diff
@@ -445,23 +445,14 @@
      */
     void registerTransform(Action<? super VariantTransform> registrationAction);
 
-    /**
-     * Registers an artifact transform with a parameter object.
-     *
-     * @see TransformAction
-     * @since 5.2
-     */
-    @Incubating
-    <T> void registerTransform(Class<T> parameterType, Action<? super ParameterizedTransformSpec<T>> registrationAction);
-
     /**
      * Registers an artifact transform without a parameter object.
      *
      * @see TransformAction
      * @since 5.3
      */
     @Incubating
-    <T extends TransformAction> void registerTransformAction(Class<T> actionType, Action<? super TransformSpec> registrationAction);
+    <T extends TransformParameters> void registerTransform(Class<? extends TransformAction<T>> actionType, Action<? super TransformSpec<T>> registrationAction);
```

Comment:
These changes won't break AGP, because it only uses `registerTransform(Class, Action)`, and that's staying the same, right?

Reply:
AGP uses `registerTransform(Action)`. The whole new register API will only be present in 5.3.